CONTRIBUTING.md

```diff
@@ -0,0 +1,117 @@
+# Contributing
+
+When contributing to Figura, please first discuss and/or interact with our community via our [Discord](https://discord.gg/figuramc). 
+
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+
+## Pull Request Process
+note - do not merge upstream into your pr branch if a clean rebase is possible
+
+1. Ensure the feature is In demand via the `📧丨mod-suggestions` channel in our [Discord](https://discord.gg/figuramc) or has the [Enhancment]([https://github.com/FiguraMC/Figura/labels/enhancement](https://github.com/FiguraMC/Figura/issues?q=is%3Aissue+is%3Aopen+label%3Aenhancement+)) tag in the Issues tab 
+   
+    OR:
+
+   Ensure what you are doing is actually a bug via the `👾丨bug-reports` channel in our [Discord](https://discord.gg/figuramc) or has the [Bug]([https://github.com/FiguraMC/Figura/issues](https://github.com/FiguraMC/Figura/issues?q=is%3Aissue+is%3Aopen+label%3Abug+)) tag in the Issues tab 
+
+3. Communicate with Devs and other contributors about what you are building and adding. This helps keep multiple people from working on the same thing alone.
+
+4. Clearly and extensively document everything you've added/changed in the pull request. This helps when compiling all the changes into full release patch notes.
```

Comment:
I think "Provide a clear list of changes, fixes, and potential/actual regressions introduced by your PR to help with making change log for the next release" would be better, as currently it is implied that almost every line needs to have a comment (which is, while helpful to people just studying off the Figura codebase (which I doubt is many because of complexity of the mod), may not be that beneficial to other devs and creates unnecessary code clutter).